app/src/main/java/com/poupa/vinylmusicplayer/service/MusicService.java

```diff
@@ -394,6 +394,13 @@ public synchronized void restoreQueuesAndPositionIfNecessary() {
             if (restoredQueue.size() > 0 && restoredQueue.size() == restoredOriginalQueue.size() && restoredPosition != -1) {
                 playingQueue = new StaticPlayingQueue(restoredQueue, restoredOriginalQueue, restoredPosition, playingQueue.getShuffleMode());
 
+                // Adjust for removed songs, marked with Song.EMPTY in the restored queues
+                for (int i = restoredQueue.size() - 1; i >= 0; --i) {
+                    if (restoredQueue.get(i).id == Song.EMPTY_SONG.id) {
+                        playingQueue.remove(i);
+                    }
+                }
```

Comment:
Shouldn't you put this for loop inside the StaticPlayingQueue constructor to ensure that a queue can never have empty song in it ?

Reply:
In fact StaticPlayingQueue (or any future queue implem) should not need to know about this anomaly due to the orphan entries. The lists presented to the queue implem should be clean, and that's what I have attempted in previous commits.

The issue is that the queues contents and the queue current position should be loaded/saved together, and in this case, adjusted together. This sould require heavy rewrite of the queue persistency logic (now spread between preference and DB).

Or they are presented to the queue implem unadjusted, and then we rely on the queue logic to "atomically" adjust them post-constructor.

If this reasoning is acceptable, I can amend the code to add the above comments to add clarity

Reply:
My logic was that the queue cannot show an empty_song, thus can manage this case as it could append on other occasion (however in this case add operation inside the queue should check for empty_song)

Reply:
OK, I'll look into doing the cleanup/adjustment before calling the queue constructor.

> My logic was that the queue cannot show an empty_song, 

It actually can handle empty_song (or any orphan song), because right now the existence of an underlying song it not required. If a song in the queue is removed from the mediastore/discog, the queue is not updated and the playing will eventually gets stuck when it reaches that empty/removed song.

Nevertheless, as said, I'll have a look in doing the cleanup before hand.

Reply:
Changed my mind and I've put the cleanup code in the ctor like you suggested - via 9acb127